sdk/include/locks.hh

```diff
@@ -337,5 +344,26 @@ class LockGuard
 			wrappedLock->unlock();
 		}
 	}
+
+	/**
+	 * Conversion to bool.  Returns true if this guard owns the lock, false
+	 * otherwise.  This allows lock guards to be used with a timeout in
+	 * conditional blocks, such as:
+	 *
+	 * ```
+	 * if (LockGuard g{lock, timeout})
```

Comment:
The guard object is in scope until the end of the `if` and `else` blocks or the end of the enclosing scope?

Reply:
Guards are in scope for the duration of the statement (if statements take a block as a body).  They are not in scope in the `else` clause.